atlas/query_node_API_sketch.graphql

```diff
@@ -0,0 +1,196 @@
+
+enum Language {
+  Chinese
+  English
+  Arabic
+  Portugese
+  French
+  # ...
+}
+
+type Channel {
+
+  id: ID!
+
+  # Id of underlying entity.
+  entityID: BigInteger!
+
+  owner: Member!
+
+  handle: String!
+
+  description: String!
+
+  coverPhotoURL: String!
+
+  avatarPhotoURL: String!
+
+  isPublic: Bool!
+
+  isCurated: Boolean!
+
+  language: Language
+
+  videos: [Video!]
+}
+
+type Category {
+
+  id: ID!
+
+  # Id of underlying entity.
+  entityID: BigInteger!
+
+  name: String!
+
+  videos: [Video!]
+}
+
+type JoystreamVideoMediaLocation {
+  dataObjectID: BigInteger!
+}
+
+type HTTPVideoMediaLocation {
+  host: String!
+  port: Int
+}
+
+# In the future we can add IPFS, Dat, etc.
+union MediaLocation = JoystreamMediaLocation | HTTPMediaLocation
+
+# Mixed both encoding and containers, only having popular combos, may need to be changed later.
+enum VideoMediaEncoding {
+  H264_mpeg4
+  VP8_WEBM
+  Theroa_Vorbis
+}
+
+# Apparently there are lots of different Creative Commons licenses,
+# read about all here https://creativecommons.org/licenses/,
+# I haven't had the time.
+enum CreativeCommonsVersion {
+  CC_BY
+  CC_BY_SA
+  CC_BY_ND
+  CC_BY_NC
+  CC_BY_NC_SA
+  CC_BY_NC_ND
+}
+
+type CreativeCommonsLicense {
+  version:
+}
+
+type UserDefinedLicense {
+  text: String!
+}
+
+union License = UserDefinedLicense | CreativeCommonsLicense
+
+type VideoMedia {
+
+  id: ID!
+
+  # Id of underlying entity.
+  entityID: BigInteger!
+
+  encoding: VideoMediaEncoding!
+
+  # Resolution width
+  pixelWidth: Int!
+
+  # Resolution height
+  pixelHeight: Int!
+
+  # Size in bytes
+  size: BigInteger
+
+  # where to find
+  location: MediaLocation!
+}
+
+type Video {
+
+  id: ID!
+
+  # Id of underlying entity.
+  entityID: BigInteger!
+
+  channel: Channel!
+
+  category: Category!
+
+  title: String!
+
+  description: String!
+
+  # In seconds
+  duration: Int!
+
+  # In intro
+  skippableIntroDuration: Int
+
+  thumbnailURL: String!
+
+  Language: Language
+
+  media: VideoMedia!
+
+  hasMarketing: Boolean
+
+  # Timestamp of block
+  publishedOnJoystreamAtblockHeight: BigInteger!
+  publishedOnJoystreamAtTimeStamp: BigInteger!
+
+  # Possible time when video was published before Joystream
+  publishedBeforeJoystream: DateTime
+
+  isPublic: Bool!
+
+  isCurated: Boolean!
+
+  isExplicit: Boolean!
+
+  license: License!
+}
+```
+
+## Queries
+
+```graphql
+
+union FreeTextSearchResultItemType = Video | Channel
+
+type FreeTextSearchResult {
+
+   item: FreeTextSearchResultItemType!
+
+   rank: Int!
+}
+
+type Query {
+
+  # Lookup a channel by its ID
+  channel(id: ID!): Channel
+
+  # List all channel by given constraints
+  channels(order_by_creation_date: Boolean, ignore_curated: Boolean, ignore_non_public: Boolean, offset: Int, limit: Int): [Channel!]
```

Comment:
You are correct.

With the current encoding, you could end up even having contradictory specifications about how to order, you could have `order_by_x` and `order_by_y` both be true, which is not semantically meaningful. The API we will get will follow the `OpenCRUD` GraphQL standard, at least in spirit, you can check it out here https://www.opencrud.org/.

In that system, there would be exactly the solution you suggested, see here

https://www.opencrud.org/#sec-Queries-Top-level.Example

I think the remedy here would be to get you an example of this from the query node (Hydra), so you can get it exactly right because my memory is obviously a poor guide.